src/main/java/com/google/firebase/auth/OidcProviderConfig.java

```diff
@@ -113,6 +143,43 @@ public CreateRequest setIssuer(String issuer) {
       return this;
     }
 
+    /**
+     * Sets whether to enable the code response flow for the new provider. By default, this is not
+     * enabled if no response type is specified.
+     *
+     * <p>A client secret must be set for this response type.
+     *
+     * <p>Having both the code and ID token response flows is currently not supported.
+     *
+     * @param enabled A boolean signifying whether the code response type is supported.
+     */
+    public CreateRequest setCodeResponseType(boolean enabled) {
+      if (properties.get("responseType") == null) {
```

Comment:
You can simplify this and reduce some of the duplication by introducing a helper:

```
private Map<String, Boolean> ensureResponseType() {
    if (properties.get("responseType") == null) {
        properties.put("responseType", new HashMap<String, Boolean>());
    }

   return (Map<String, Boolean>) properties.get("responseType");
}
```